compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -304,6 +305,18 @@ impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for Span {
     }
 }
 
+impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for Symbol {
+    fn encode(&self, s: &mut EncodeContext<'a, 'tcx>) {
+        if !s.symbol_table.contains_key(self) {
+            let x = s.lazy(self.to_string());
```

Comment:
Ok, this looks simpler, but fail on first write here:
```
thread 'rustc' panicked at 'assertion failed: `(left == right)`
  left: `NodeStart(30)`,
 right: `NoNode`', compiler\rustc_metadata\src\rmeta\encoder.rs:397:9

```

Reply:
`encode_crate_root` calls `encode_crate_deps` that calls `self.lazy_array`
https://github.com/rust-lang/rust/blob/53792b9c5c3cded466a073e4b9a9e51d0007b9ab/compiler/rustc_metadata/src/rmeta/encoder.rs#L1762

that change state to `LazyState::NodeStart`
https://github.com/rust-lang/rust/blob/53792b9c5c3cded466a073e4b9a9e51d0007b9ab/compiler/rustc_metadata/src/rmeta/encoder.rs#L394-L406

Where it founds `Symbol` and tries to encode it but can't, because of wrong state.